les/vflux/client/fillset_test.go

```diff
@@ -28,12 +28,16 @@ import (
 )
 
 type testIter struct {
+	closed bool
 	waitCh chan struct{}
 	nodeCh chan *enode.Node
 	node   *enode.Node
 }
 
 func (i *testIter) Next() bool {
+	if i.closed {
+		return false
+	}
```

Comment:
I don't know. This introduces a new field, which you access from potentially two separate threads without synchronization. 
So while this PR fixes a leak, it also IMO introduces new bugs -- either a potential send over close channel, or a potential simultaneous read/write.

Reply:
Thank you, you are right. `testIter` need mutex protect if it is used in concurrent scenario.
But I revisit `fillset_test.go` and `fillset.go`, `testIter.closed` field was  written by `testIter.Close()` was called safely in `TestFillSet()` only once at here:
https://github.com/ethereum/go-ethereum/blob/master/les/vflux/client/fillset_test.go#L111
and `testIter.closed` field was read by `testIter.Next()` in a single goroutine (`(fs *FillSet) readLoop()`) at here:
https://github.com/ethereum/go-ethereum/blob/master/les/vflux/client/fillset.go#L64
so, after `testIter.Close()` was called, send to `waitCh` won't happen.